integration/integration_test.go

```diff
@@ -6547,6 +6548,42 @@ func testListResourcesAcrossClusters(t *testing.T, suite *integrationTestSuite)
 	}
 }
 
+func testJoinOverReverseTunnelOnly(t *testing.T, suite *integrationTestSuite) {
+	lib.SetInsecureDevMode(true)
+	defer lib.SetInsecureDevMode(false)
+
+	// Create a Teleport instance with Auth/Proxy.
+	mainConfig := func() *service.Config {
```

Comment:
Why do we need the wrapping func? Could we do `mainConfig := suite.defaultServiceConfig()` and take it from there?